packages/esm-patient-orders-app/src/order-basket-action-button/order-basket-action-button.extension.tsx

```diff
@@ -1,21 +1,42 @@
 import React, { type ComponentProps } from 'react';
 import { useTranslation } from 'react-i18next';
-import { ActionMenuButton, ShoppingCartIcon } from '@openmrs/esm-framework';
+import { ActionMenuButton, ShoppingCartIcon, useWorkspaceContext } from '@openmrs/esm-framework';
 import { useLaunchWorkspaceRequiringVisit, useOrderBasket } from '@openmrs/esm-patient-common-lib';
 
 const OrderBasketActionButton: React.FC = () => {
   const { t } = useTranslation();
   const { orders } = useOrderBasket();
+  const { activeWorkspace } = useWorkspaceContext();
   const launchOrderBasket = useLaunchWorkspaceRequiringVisit('order-basket');
 
+  const isWorkspaceOpen = activeWorkspace?.workspace === 'order-basket';
+
+  const handleClick = () => {
+    if (!isWorkspaceOpen) {
+      launchOrderBasket();
+    }
+  };
+
+  const getIcon = (props: ComponentProps<typeof ShoppingCartIcon>) => (
+    <span
+      onClick={(e) => {
+        if (isWorkspaceOpen) {
+          e.stopPropagation();
+        }
```

Comment:
Do we need this kind of guard here? Isn't it enough to just change the `handleClick()` stuff?